### PR TITLE
Add index validation

### PR DESCRIPTION
**Pull request by ExcaliburZero**
Add index validation to the methods of the Sheet, Row, and Column classes. With these changes, if you pass a 0 or negative value as a row or column index then a RangeError is thrown.

```
> sheet.cell(0, 1);
RangeError: Invaid row index 0. Remember that spreadsheets use 1 indexing.
```

This helps to prevent invalid spreadsheets that could be successfully generated, but be rejected by Excel as being corrupted (see #111).

I wrote a bunch of test cases for different places that invalid indices could be passed in, let me know if there are any other cases I should cover.

Also some of the test cases seem to be failing oddly, see `Sheet activeCell should throw an exception on a row number of 0`. This should lead to `sheet.activeCell` calling `this.cell` which should then be performing the validation on the row index, however the exception is not being thrown in the test. When I actually compile the code with gulp and run similar test in my browser it does throw the exception.

```
> XlsxPopulate.fromBlankAsync().then(workbook => workbook.sheet(0).activeCell(0, 1));
RangeError: Invaid row index 0. Remember that spreadsheets use 1 indexing.
```

### CODE REVIEW

File: lib/Sheet.js [hunk 1]
```diff
@@ -358,6 +363,8 @@ class Sheet {
                 return new Range(startCell, endCell);
             })
             .case(['number', '*', 'number', '*'], (startRowNumber, startColumnNameOrNumber, endRowNumber, endColumnNameOrNumber) => {
+                indexValidation.validateIndex(startRowNumber, "row");
+                indexValidation.validateIndex(endRowNumber, "row");
```
Comment (eddiecorrigall) [inline lib/Sheet.js]: You need to validate both row range AND column range, that is why your tests are failing in Sheet.spec.js.

**Comment by eddiecorrigall** [comment]
Hi, thank you for contributing to the project. I'd like to merge this code, if you are willing to keep working on this pull request. Please first rebase your branch and resolve any conflicts. Ill see if I can help you resolve the failed tests.

File: lib/Sheet.js (hunk 1 above)
Comment (ExcaliburZero) [reply lib/Sheet.js]: For this case I thought that on line 368 the `cell` method of the `Sheet` would be called, which would in turn call the `cell` method on a specific `Row`, which has index validation for the column number on `lib/Row.js:53`.

I wanted to try to keep the validation lower if possible.

File: lib/Sheet.js [hunk 2]
```diff
@@ -84,6 +85,8 @@ class Sheet {
                 return this.cell(cellAddress);
             })
             .case(['number', '*'], (rowNumber, columnNameOrNumber) => {
+                indexValidation.validateIndex(rowNumber, "row");
```
Comment (eddiecorrigall) [inline lib/Sheet.js]: This is unnecessary since Sheet.cell will validate

File: lib/Sheet.js [hunk 3]
```diff
@@ -120,9 +123,11 @@ class Sheet {
             .case('string', address => {
                 const ref = addressConverter.fromAddress(address);
                 if (ref.type !== 'cell') throw new Error('Sheet.cell: Invalid address.');
+                indexValidation.validateIndex(ref.rowNumber, "row");
```
Comment (eddiecorrigall) [inline lib/Sheet.js]: Unnecessary

File: lib/Sheet.js [hunk 4]
```diff
@@ -120,9 +123,11 @@ class Sheet {
...
                 return this.row(ref.rowNumber).cell(ref.columnNumber);
             })
             .case(['number', '*'], (rowNumber, columnNameOrNumber) => {
+                indexValidation.validateIndex(rowNumber, "row");
```
Comment (eddiecorrigall) [inline lib/Sheet.js]: Unnecessary

File: lib/Sheet.js (hunk 2 above)
Comment (ExcaliburZero) [reply lib/Sheet.js]: I have now removed this unnecessary validation.

File: lib/Sheet.js (hunk 1 above)
Comment (eddiecorrigall) [reply lib/Sheet.js]: I missed that. But I did see that the column validation test was failing, that why I brought it up.

**Pull request merged**